### PR TITLE
cache search result to fix rerequest seach string after export or remove

### PR DESCRIPTION
**Pull request by hawkeyexp**
To prevent from a new input dialog after export or remove to/from db in the list of search results let's cache the search string (and reuse it on refresh which is called after export or remove) until the main menu is reloaded.

### CODE REVIEW

File: resources/lib/KodiHelper.py [hunk 1]
```diff
@@ -87,12 +87,31 @@ def show_search_term_dialog (self):
         :obj:`str`
             Term to search for
         """
-        dlg = xbmcgui.Dialog()
-        term = dlg.input(self.get_local_string(string_id=30003), type=xbmcgui.INPUT_ALPHANUM)
-        if len(term) == 0:
-            term = ' '
+        term = xbmcgui.Window(xbmcgui.getCurrentWindowId()).getProperty('searchstring')
```
Comment (asciidisco) [inline resources/lib/KodiHelper.py]: Maybe the existing cache helper methods could be reused? https://github.com/asciidisco/plugin.video.netflix/blob/master/resources/lib/KodiHelper.py#L286-L327

Comment (hawkeyexp) [reply resources/lib/KodiHelper.py]: I have some issues caused by cachhelper so it should be rechecked if it makes sense in current version (not refreshing lists etc.) and the search string is only a single string so i have made it standalone..

**Comment by hawkeyexp** [comment]
Will try to build a better integration

**Comment by asciidisco** [comment]
Not necessarily a bad implementation & also a good thought.
We should may figure out what´s wrong with the cache helper methods/or with the usage of´em, so that they can be used for this. I´d like to reopen it, as it is close to a working solution.

**Comment by hawkeyexp** [comment]
Caching changed to your memcache - seems to work. Only one thing should be added: when search result is shown and navigating back search cache should be cleaned - back action triggers no rebuild of previous menu so cache is not cleaned. Possible you can add it.

**Comment by hawkeyexp** [comment]
Don't merge - seems to be buggy again

**Comment by hawkeyexp** [comment]
Seems like pull "Improve search and error handling #151" solves the problem and makes this obsolete.

**Pull request closed without merging**